py/torch_tensorrt/dynamo/_refit.py

```diff
@@ -0,0 +1,335 @@
+        settings=restored_settings,
+    )
+    return new_trt_gm
+
+
+def _refit_module_weights(
+    compiled_module: torch.fx.GraphModule | ExportedProgram,
+    new_weight_module: ExportedProgram,
+    inputs: Tuple[Any, ...],
+    settings: Any,
+) -> torch.fx.GraphModule:
+    """
+    Refit a compiled graph module with ExportedProgram
+    """
+
+    if settings.debug:
+        set_log_level(logger.parent, logging.DEBUG)
+
+    if not isinstance(inputs, collections.abc.Sequence):
+        inputs = [inputs]
+
+    if isinstance(compiled_module, ExportedProgram):
+        compiled_module = compiled_module.module()
+
+    # Prepare torch_trt inputs
+    inputs = prepare_inputs(inputs)
+    device = to_torch_tensorrt_device(settings.device)
+    runtime = trt.Runtime(TRT_LOGGER)
+    if not isinstance(new_weight_module, ExportedProgram):
+        raise AssertionError(
+            f"Input graph should be an ExportedProgram but got type {type(new_weight_module)}"
+        )
+    new_weight_module = new_weight_module.run_decompositions(
+        get_decompositions(settings.enable_experimental_decompositions)
+    )
+    gm = new_weight_module.module()
+    logger.debug("Input graph: " + str(gm.graph))
+    # Apply lowering on the graph module
+    torch_inputs = get_torch_inputs(inputs, device)
+    gm = apply_lowering_passes(gm, torch_inputs)
+
+    logger.info("Compilation Settings: %s\n", settings)
+
+    # Set torch-executed ops
+    CONVERTERS.set_disallowed_targets(settings.torch_executed_ops)
+
+    # If specified, try using the fast partitioner and fall back to the global one on failure
+    if settings.use_fast_partitioner:
+        try:
+            partitioned_module, supported_ops = partitioning.fast_partition(
+                gm,
+                verbose=settings.debug,
+                min_block_size=settings.min_block_size,
+                torch_executed_ops=settings.torch_executed_ops,
+            )
+        except torch.fx.passes.splitter_base.FxNetSplitterInternalError:
+            logger.error(
+                "Partitioning failed on the subgraph with fast partition. See trace above. "
+                + "Retrying with global partition.",
+                exc_info=True,
+            )
+
+            fast_partitioner_failed = True
+            settings.use_fast_partitioner = False
+
+    if not settings.use_fast_partitioner:
+        partitioned_module, supported_ops = partitioning.global_partition(
+            gm,
+            verbose=settings.debug,
+            min_block_size=settings.min_block_size,
+            torch_executed_ops=settings.torch_executed_ops,
+        )
+    # PytorchTensorRTModule does not support deepcopy
+    # Create a shallow copy. Replace the TRTModule after
+    compiled_module = copy.copy(compiled_module)
+    # Iterate over all components that can be accelerated
+    # Generate the corresponding TRT Module for those
+    for name, _ in partitioned_module.named_children():
```

Comment:
We need to ensure that the new module's partition is the same as the compiled modules partition. 

1. We can check the number of subgraphs, perhaps the names as well (if deterministic) 
2. At compile time, compute the hash of the source fx graph (https://github.com/pytorch/pytorch/blob/fba21edf5b9aa14babb9c0bc860dc9c597eb8010/torch/_inductor/codecache.py#L670) and store as attribute in the TRTModule. The compare the hash of the new graph to the one stored in the compiled subgraph module

Reply:
@zewenli98 You might be interested in reusing this part for engine caching